luncheon/request.lua

```diff
@@ -17,9 +15,9 @@ local shared = require "luncheon.shared"
 ---@field private _source fun(pat:string|number|nil):string
 ---@field private _parsed_headers boolean
 ---@field private _received_body boolean
----@field public mode Mode How this request was built
 ---@field public trailers Headers|nil The HTTP trailers
 local Request = {}
+setmetatable(Request, ReqResp)
```

Comment:
I'm not the strongest on OOP in Lua quite yet and haven't really dabbled with multiple inheritance too much. But I'm more used to seeing patterns where `setmetatable` is called on an `instance` of a class.

I'd like @FreeMasen to weigh in if he has time, but my gut says it might make more sense to use `Request.__index = ReqResp` and then use `setmetatable` on instances of `Request`. Same with `Response`. But I could also be missing something entirely here as well that makes that the wrong choice; maybe that makes `.source()` and other stuff that call through to "super" methods trickier? It's not clear to me, so again I might be totally off base here.

Reply:
There seems to be a few ways to setup class definitions, and constructors but only one way for it to work. I tried to follow the [example in the lua docs](https://www.lua.org/pil/16.2.html), but this is the only way I could get it without changing the Request and Response constructor apis.